src/components/webcam-texture-target.js

```diff
@@ -0,0 +1,76 @@
+import { disposeTexture } from "../utils/material-utils";
+import { createVideoOrAudioEl } from "../utils/media-utils";
+
+AFRAME.registerComponent("webcam-texture-target", {
+  schema: {
+    src: { type: "string" }
+  },
+
+  getMaterial() {
+    return (
+      (this.el.object3DMap.skinnedmesh && this.el.object3DMap.skinnedmesh.material) ||
+      (this.el.object3DMap.mesh && this.el.object3DMap.mesh.material) ||
+      this.el.object3D.material
+    );
+  },
+
+  init() {
+    const material = this.getMaterial();
+    this.originalTexture = material && material.map;
+  },
+
+  update(prevData) {
+    const material = this.getMaterial();
+
+    console.log({ src: this.data.src, material, prevSrc: prevData.src });
+
+    if (!material) {
+      return;
+    }
+
+    const src = this.data.src;
+
+    if (prevData.src === src) {
+      return;
+    }
+
+    if (src && src.startsWith("hubs://")) {
+      const streamClientId = src.substring(7).split("/")[1]; // /clients/<client id>/video is only URL for now
+
+      this.originalTexture = material.map;
```

Comment:
Why is `originalTexture` reassigned here?

Reply:
Yeah, I'm going to remove this. The component should have the precondition that the entity has a mesh with a material prior to being added. I've added a warning and another check in the remove function to guard against that.